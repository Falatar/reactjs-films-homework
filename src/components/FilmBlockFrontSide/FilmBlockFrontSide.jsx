import React from 'react';
import PropTypes from 'prop-types';
import style from './FilmBlockFrontSide.scss';
import FilmDescription from '../FilmDescription';
import play from '../../static/play.svg';

function FilmBlockFrontSide({
  id, name, tagList, rating, callModal, switchMode, poster,
}) {
  return (
    <div className={style.film__block}>
      <div className={style.control}>
        <img
          className={style.wall}
          src={poster}
          alt={id}
        />
        <div className={style.hidden}>
          <button type="button" className={style.play} onClick={callModal}>
            <svg className={style.svg}>
              <use xlinkHref={play} />
            </svg>
          </button>
          <p className={style.watch__now}>Watch Now</p>
          <button type="button" className={style.info} onClick={switchMode}>View Info</button>
        </div>
      </div>
      <FilmDescription
        name={name}
        tagList={tagList}
        rating={rating}
        mode
      />
    </div>
  );
}

FilmBlockFrontSide.defaultProps = {
  id: 0,
  name: 'Can\'t find property "name"',
  poster: '',
  rating: 0,
  tagList: 'Can\'t find property "tagList"',
  callModal: () => 'can\'t find callModal function',
  switchMode: () => 'can\'t find switchMode function',
};

FilmBlockFrontSide.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  poster: PropTypes.string,
  rating: PropTypes.number,
  tagList: PropTypes.string,
  callModal: PropTypes.func,
  switchMode: PropTypes.func,
};

export default FilmBlockFrontSide;
