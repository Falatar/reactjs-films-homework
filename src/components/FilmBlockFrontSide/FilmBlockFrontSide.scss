.film__block {

    height: 35vh;
    width: 46vh;
    background-color: white;
    margin: 1%;
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    background-color: white;
    background-size: cover;

    .control {

        width: 100%;
        height: 80%;
        margin: 0;
        padding: 0;
        border: 0;

        .wall {
            display: block;
            width: 100%;
            height: 100%;
            background-size: cover;
            margin: 0;
            padding: 0;
            border: 0;
            z-index: 100;
            position: relative;
        }

        .wall:hover {
            z-index: 1;
        }

        .hidden:hover {
            z-index: 101;
        }

        .hidden {

            background-color: rgba(0, 0, 0, 0.664);
            width: 100%;
            height: 100%;
            top: -100%;
            margin: 0;
            padding: 0;
            border: 0;
            position: relative;
            z-index: 50;
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: center;

            .play {
                z-index: inherit;
                color: white;
                background-color: deepskyblue;
                height: 1.5em;
                width: 1.5em;
                border: 0;
                border-radius: 100%;
                font-size: 3.5em;
                outline: none;
                padding: 0;
                margin-top: 10%;

                .svg {
                    width: 100%;
                    height: 100%;
                    fill: white;
                    padding-left: 7%;
                }
            }

            .play:active {
                background-image: -webkit-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
                background-image: -moz-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
                background-image: -o-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
                background-image: -ms-linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
                background-image: linear-gradient(bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.4));
            }

            .watch__now {
                color: white;
                font-weight: 500;
                font-size: 1.15em;
            }

            .info {
                //margin-top: 5%;
                margin-bottom: 5%;
                background: transparent;
                color: white;
                font-size: 1.15em;
                border: 1px solid;
                padding: 2% 5%;
                border-radius: 0.5em;
                outline: none;
            }

            .info:active {
                background: rgba(0, 0, 0, 0.9);
            }
        }
    }
}