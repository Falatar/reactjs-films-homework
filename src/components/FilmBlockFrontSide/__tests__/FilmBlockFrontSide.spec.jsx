import React from 'react';
import renderer from 'react-test-renderer';
import FilmBlockFrontSide from '../FilmBlockFrontSide';

jest.mock('../../FilmDescription', () => 'FilmDescription');

describe('FilmBlockFrontSide', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <FilmBlockFrontSide
          poster=""
          id={1}
          name="nemo"
          tagList="empty"
          rating={0.0}
          callModal={() => {}}
          switchMode={() => {}}
          checkImgSizes={() => {}}
          setDefaultImg={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('default functions works correctly', () => {
    const component = renderer.create(
      <FilmBlockFrontSide />,
    );
    const result1 = component.root.props.callModal();
    const result2 = component.root.props.switchMode();
    expect(result1).toEqual('can\'t find callModal function');
    expect(result2).toEqual('can\'t find switchMode function');
  });
});
