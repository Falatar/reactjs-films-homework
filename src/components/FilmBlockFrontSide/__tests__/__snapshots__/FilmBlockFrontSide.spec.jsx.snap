// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmBlockFrontSide renders correctly 1`] = `
<div
  className="film__block"
>
  <div
    className="control"
  >
    <img
      alt={1}
      className="wall"
      src=""
    />
    <div
      className="hidden"
    >
      <button
        className="play"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="svg"
        >
          <use
            xlinkHref="mocked-image"
          />
        </svg>
      </button>
      <p
        className="watch__now"
      >
        Watch Now
      </p>
      <button
        className="info"
        onClick={[Function]}
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
  <FilmDescription
    mode={true}
    name="nemo"
    rating={0}
    tagList="empty"
  />
</div>
`;
